.github/workflows/auto-approve.yml

```diff
@@ -0,0 +1,19 @@
+name: Auto-approve
+
+on:
+  pull_request_target:
+    branches:
+      - main
+
+jobs:
+  auto-approve:
+    name: Dependency bump
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```

Comment:
This is a potential vulnerability, any branch in `deps/` will be auto-approved, would guard against that by also checking that the PR author is `hash-worker[bot]`.

Reply:
I have noted that already in the known-issues:
> We might need to further restrict the auto-approval constraints, but for now this should be good enough

The worker currently is _only_ used for creating commits/open PRs, so this is not a concern, yet. We also need to restrict the event to precisely say when an approval should happen. Ideally, only if CI is green.
```suggestion
    # Further restrict this to
    # - Check the PR author (not only the actor)
    # - Check the event (only commits should trigger auto-approvals)
    # - (optional) check for green CI
    if: github.actor == 'hash-worker[bot]' && startsWith(github.head_ref, 'deps/')
```